src/Databases/MySQL/MaterializeMetadata.cpp

```diff
@@ -293,9 +294,39 @@ void MaterializeMetadata::startReplication(
     try
     {
         connection->query("FLUSH TABLES;").execute();
-        connection->query("FLUSH TABLES WITH READ LOCK;").execute();
 
+        if (materialized_tables_list.empty())
+        {
+            /// "FLUSH TABLES WITH READ LOCK" requires the SUPER privilege, which in RDS and some managed databases
+            /// is not supported even for the master user. However, we can use FLUSH TABLES table_names WITH READ LOCK.
+            std::unordered_set<String> base_tables_list;
+
+            /// Get only base tables (not views).
+            mysqlxx::UseQueryResult result_list = connection->query("SHOW FULL TABLES IN " + database + " WHERE Table_type = 'BASE TABLE'").use();
+
+            while (mysqlxx::Row row = result_list.fetch())
+                base_tables_list.insert(row[0].getString());
+
+            connection->query("FLUSH TABLES `" + boost::algorithm::join(base_tables_list, "`,`") + "` WITH READ LOCK;").execute();
```

Comment:
It's incorrect, quoting like this never works (what will happen if a table name contains backtick?). Use `backQuoteIfNeed`.